dsl/pace/dsl/dace/dace_config.py

```diff
@@ -112,7 +117,11 @@ def __init__(
                 "args",
                 value="-std=c++14 -Xcompiler -fPIC -O3 -Xcompiler -march=native",
             )
-            dace.config.Config.set("compiler", "cuda", "cuda_arch", value="60")
+
+            cuda_sm = 60
+            if cp:
+                cuda_sm = cp.cuda.Device(0).compute_capability
+            dace.config.Config.set("compiler", "cuda", "cuda_arch", value=f"{cuda_sm}")
```

Comment:
I think you are querying the device to understand its particular architecture, so my question follows from that.  Is the default_block_size value always going to be _"64,8,1"_, or is there a query below that resets it?  If not, maybe some description of why this default.

Reply:
I shall add a comment on the block size default.

Basically if `sm` relates to the feature the code will be injected with (and the card can handle), block size depends on the actual architecture you are using. `64,8` will work for Pascal forward, but become slightly overpowered with the years. The true way of addressing this challenge is running a brute-force learner to actually test per kernel what the configuration is more optimal. I will bring this feature to Pace next.